### ISSUE TITLE
vsc points to DIAGNOSTIC_SEVERITY_RULES.md which is deprecated

### ISSUE THREAD

**Issue by fenchu** [microsoft/pylance-release#3589]
pylance==v2022.11.10

Pylance problems in vsc points to a deprecated file, and when you go to the new file:
https://github.com/microsoft/pyright/blob/main/docs/configuration.md#type-check-diagnostics-settings (anchor do not work)

https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md missing
```
[{
	"resource": "/c:/dist/work/hso-fullstack-test/tests/test_10_phone_registration_en.py",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"external": "https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules",
			"path": "/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "diagnostic-severity-rules"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Literal['Resources']\" cannot be assigned to parameter \"__s\" of type \"slice\" in function \"__getitem__\"\n  \"Literal['Resources']\" is incompatible with \"slice\"",
	"source": "Pylance",
	"startLineNumber": 71,
	"startColumn": 20,
	"endLineNumber": 71,
	"endColumn": 37
}]
```

**Comment by rchiodo** [comment]
Thanks for the bug.

It's not clear what the problem is? That the error points to a deprecated file?

**Comment by erictraut** [comment]
Yeah, the link points to a deprecated file. This comes from the internal method `getDocumentationUrlForDiagnosticRule`, which Pylance overrides with its own URL.

**Comment by rchiodo** [comment]
Talked this over at triage. 

We thought it would be better to:

- Fix the links to point to the pyright file (fix this issue)
- Fix the pyright file to support linking to individual entries (by adding </a> tags in the doc next to each item)
- Fix pyright to jump to those individual entries
- Remove the getDocumentationUrlForDiagnosticRule override in pylance

The original reason for this change was to not have to keep two copies of the documentation for the rules around.

@erictraut, thoughts?

**Comment by erictraut** [comment]
Yeah, that proposal makes sense to me.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221121-9-November-2022-PreRelease)
